### PR TITLE
Make hostfxr backward compatible with hostpolicy

### PR DESCRIPTION
**Pull request by schellap**
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2392)

### CODE REVIEW

**Comment by schellap** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please

**Comment by schellap** [comment]
@gkhanna79, PTAL. Makes the interface stable between the two dlls.

**Comment by schellap** [comment]
@brthor for the non native changes, PTAL. Makes the arg order consistent.

**Comment by brthor** [comment]
NICE :shipit:

File: src/corehost/cli/fxr/fx_muxer.cpp [hunk 1]
```diff
@@ -254,60 +254,66 @@ int muxer_usage()
...
+int fx_muxer_t::parse_args_and_execute(
+    const pal::string_t& own_dir,
+    const pal::string_t& own_dll,
+    int argoff, int argc, const pal::char_t* argv[], bool exec_mode, host_mode_t mode, bool* is_an_app)
 {
     *is_an_app = true;
 
     std::vector<pal::string_t> known_opts = { _X("--additionalprobingpath") };
-    if (exec_mode)
+    if (exec_mode || mode == host_mode_t::standalone)
```
Comment (gkhanna79) [inline src/corehost/cli/fxr/fx_muxer.cpp]: Is this for the unpublished standalone app?

**Comment by pakrym** [comment]
Can we get this in? Blocking aspnet and nuget

**Comment by gkhanna79** [comment]
@schellap Can you please resolve the conflicts to prioritize getting this in?

@pakrym Once @schellap completes the testing, he will get the change in tonight.

**Comment by gkhanna79** [comment]
Windows builds failed with identical issue:

20:08:32 [EXEC       <] [FAIL] [00:01:55.097] C:\Users\dotnet-bot\.nuget\packages\runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR\1.0.2-rc2-24008\tools\crossgen.exe "-readytorun" "-in" "D:\j\workspace\release_windo---4c3b899b\artifacts\win81-x64\stage1\shared\Microsoft.NETCore.App\1.0.0-rc2-3002347\Microsoft.CodeAnalysis.CSharp.dll" "-out" "D:\j\workspace\release_windo---4c3b899b\artifacts\win81-x64\stage1\shared\Microsoft.NETCore.App\1.0.0-rc2-3002347\Microsoft.CodeAnalysis.CSharp.readytorun" "-platform_assemblies_paths" "D:\j\workspace\release_windo---4c3b899b\artifacts\win81-x64\stage1\shared\Microsoft.NETCore.App\1.0.0-rc2-3002347" exited with -2146234279

@brthor @piotrpMSFT Is this known?

**Comment by gkhanna79** [comment]
The error code corresponds to 0x80131059 that corresponds to runtime/NI mismatch for mscorlib.ni.dll (see https://github.com/dotnet/coreclr/blob/master/src/inc/corerror.xml#L575).

Here is the subsequent excerpt from the log:

20:08:17 Standard Error:
20:08:17 Could not load file or assembly 'D:\j\workspace\debug_windows---54dcb9ec\artifacts\win81-x86\stage1\shared\Microsoft.NETCore.App\1.0.0-rc2-3002347\mscorlib.dll' or one of its dependencies. The native image could not be loaded, because it was generated for use by a different version of the runtime. (Exception from HRESULT: 0x80131059)
20:08:17 Error: compilation failed for "D:\j\workspace\debug_windows---54dcb9ec\artifacts\win81-x86\stage1\shared\Microsoft.NETCore.App\1.0.0-rc2-3002347\Microsoft.CodeAnalysis.CSharp.dll" (0x80131059)
20:08:17 

@brthor Which CoreCLR package was used to fetch crossgen/mscorlib.ni.dll for the crossgen attempt of the SharedFX in CLI build?

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Pull request merged**